app/controllers/compute_resources_vms_controller.rb

```diff
@@ -115,4 +115,17 @@ def run_vm_action(action)
     error _("Error - %{message}") % { :message => _(e.message) }
     redirect_to :back
   end
+
+  def load_vms
+    opts = {}
+    if @compute_resource.supports_vms_pagination?
+      return unless params['sEcho'].present? # We are not paginating, don't load vms
+      opts[:search] = params['sSearch'] || ''
+      opts[:max] = (params['iDisplayLength'] || 10).to_i
+      opts[:page] = (params['iDisplayStart'].to_i / opts[:max])+1
```

Comment:
Not sure, maybe if we add pagination for more providers and need to do different translation from the parameters to the options.